FOCUS #19: Sub Account ID related spec changes (#48)
FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec: specification/dimensions/subaccountid.md

Reviewer: The apendix table formatting needs corrected.

# Sub Account ID

> A sub account is an optional provider-supported construct for organizing resources and services connected to a billing account. Sub accounts are commonly used for scenarios like grouping based on organizational constructs, access management needs and cost allocation strategies. Sub accounts must be associated with a billing account as they do not receive invoices. 

Reviewer 2: ```suggestion
A sub account is an optional provider-supported construct for organizing resources and/or services connected to a billing account. Sub accounts are commonly used for scenarios like grouping based on organizational constructs, access management needs and cost allocation strategies. Sub accounts must be associated with a billing account as they do not receive invoices. 
```

> A sub account ID provider assigned identifier assigned to a sub account. It is commonly used for cost allocation, reporting, and chargeback purposes.

Reviewer 3: A sub account ID **is a** provider assigned identifier

> A sub account ID provider assigned identifier assigned to a sub account. It is commonly used for cost allocation, reporting, and chargeback purposes.

Reviewer 2: ```suggestion
A sub account ID is a provider assigned identifier assigned to a sub account.
```

> The SubAccountId column MUST be present in the billing data. This column MUST be of type String and MUST NOT contain null values. Where a provider doesn't support a sub account construct (only has a billing account), use the BillingAccountId value.

Reviewer 2: ```suggestion
The SubAccountId column MUST be present in the billing data. This column MUST be of type String. If a provider does not support a sub account construct (only has a billing account) or does support a sub account construct, but the charge does not apply to a sub account, the SubAccountId column MAY be null.
```

[Link to Appendix]()

## Column ID

SubAccountId

## Display name

Sub Account ID

## Description

> An ID assigned to a grouping of resources, often used to manage access and/or cost.

Reviewer 2: ```suggestion
An ID assigned to a grouping of resources and/or services, often used to manage access and/or cost.
```

## Content constraints

|    Constraint   |      Value      |
|:----------------|:----------------|
| Column required | True            |
| Data type       | String          |

> | Allows nulls    | False*          |

Reviewer 2: ```suggestion
| Allows nulls    | True          |
```

| Value format    | \<not specified> |

> 
> * Where a provider doesn't support a sub account construct, use the BillingAccountId value
> 

Reviewer 2: ```suggestion

```
